client/Wayland/wlf_input.c

```diff
@@ -25,22 +25,30 @@
 
 #include "wlf_input.h"
 
-BOOL wlf_handle_pointer_enter(freerdp *instance, UwacPointerEnterLeaveEvent *ev) {
-	rdpInput* input = instance->input;
+BOOL wlf_handle_pointer_enter(freerdp* instance, UwacPointerEnterLeaveEvent* ev)
+{
+	if (!instance || !ev || !instance->input)
+		return FALSE;
```

Comment:
I don't see how we can get ev == NULL, the same for instance.
The ```rdpInput* input = instance->input``` was nice for the rest of the code IMHO

Reply:
It is a callback function. Arguments should be checked and a proper error code returned. Compiler will optimize unnecessary checks anyway.
for the ```instance->input``` part, that I can do again.

Reply:
Well IMHO that's exactly the opposite, a callback is supposed to be called with some pre-requirements. Having instance != NULL and ev != NULL are the requirements, even  testing instance->input is questionable.

Look at the call hierarchy for ```wlf_handle_pointer_enter```  (Ctrl+Alt+H under eclipse) you'll see that NULL in instance or ev are just not possible. So it had some micro tests that for sure the compiler will keep as is as there's no place for optimization here. But worst, when you read the code you can be suspicious on the instance or ev parameters when there's no reason for that.